docs/parse_argparse.py

```diff
@@ -179,69 +184,67 @@ def write_usage(fpath, rst_dirpath, wrapped_lines):
         indented arparse output after running the `--help` command
     """
     # Extract the filename, eg: kmers.py
-    fname = os.path.basename(fpath)
+    basename = os.path.basename(fpath)
     # extract the "kmers" of kmers.py
-    basename = os.path.splitext(fname)[0]
-    fname_rst = basename + ".rst"  # make it kmers.rst
-    path_script_rst = os.path.join(
-        rst_dirpath, fname_rst)
-    frmt_header = "="*len(fname)
-    text = f"{frmt_header}\n{fname}\n{frmt_header}\n\n.. code-block:: shell\n"
-    with open(path_script_rst, "w") as fh_rst:
-        fh_rst.write(f"{text}\n{wrapped_lines}")
+    fname = os.path.splitext(basename)[0]
+    rst_fname = fname + ".rst"  # make it kmers.rst
+    rst_fpath = os.path.join(
+        rst_dirpath, rst_fname)
+    frmt_header = "="*len(basename)
+    header = f"{frmt_header}\n{basename}\n{frmt_header}\n\n.. code-block:: shell\n"
+    with open(rst_fpath, "w") as fh_rst:
+        fh_rst.write(f"{header}\n{wrapped_lines}")
 
 
-def write_main_file(fpath, rst_dirpath):
+def write_main_header(root, dirname):
     """
     Writes the header and basic information to all of the `main.rst` files
 
     Parameters
     ----------
-    fpath : string
-        file path that needs to be documented,
-        e.g. '</path/to/autometa/script.py>'
-    rst_dirpath : string
-        path to the directory where script.rst files will be written,
+    root : string
+        path to parent directory of the directory where `main.rst` file needs to be written
         e.g. '</path/to/docs/source/scripts/directory>'
+    dirname : string
+        name of the directory whose main file we are writing
     """
-    # extract "/home/siddharth/Autometa/autometa/common/external"
-    dir_path = os.path.dirname(os.path.abspath(fpath))
-    main_rst_dir = os.path.basename(dir_path)  # extraxt "external"
-    main_rst_file = main_rst_dir + "_main.rst"  # making a file external_main
-    main_rst_fpath = os.path.join(rst_dirpath, main_rst_file)
-    if not os.path.exists(main_rst_fpath):
-        frmt_header = "="*len(main_rst_dir)
-        outlines = f"{frmt_header}\n{main_rst_dir}\n{frmt_header}\n\n.. toctree::\n"
-        outlines += f"\t :maxdepth: 2\n\t :caption: Table of Contents\n\n"
-        with open(main_rst_fpath, "w") as fh:
-            fh.write(outlines)
-
-
-def link_main(rst_scripts_dirpath):
+    main_rst_file = dirname + "_main.rst"  # making a file external_main
+    main_rst_fpath = os.path.join(root, dirname, main_rst_file)
```

Comment:
To stick with other variables throughout the script, using `fname` for the file name reduces ambiguity. Where `file` could mean file path or file name.

```suggestion
    main_rst_fname = dirname + "_main.rst"  # making a file external_main
    main_rst_fpath = os.path.join(root, dirname, main_rst_fname)
```